src/interp.rs

```diff
@@ -299,37 +298,60 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
     /// location inside the Machine.
     fn eval_place(&mut self, place: &Place<'tcx>) -> Address {
         match *place {
-            Place::Local(local) => Address::Local(local, 0),
-            Place::Static(ref static_) => unimplemented!(),
+            Place::Local(local) => {
+                Address::Local(local, None)
+            }
             Place::Projection(ref proj) => {
                 match proj.elem {
                     ProjectionElem::Field(field, _) => {
                         let base_ty = self.place_ty(&proj.base);
-                        let offset = self.get_field_offset(base_ty, field.index());
-                        match proj.base {
-                            Place::Local(local) => {
-                                return Address::Local(local, offset);
+                        let mut ptr = self.get_field_ptr(base_ty, field.index());
+                        let base_addr = self.eval_place(&proj.base);
+                        // I'm not sure how correct this is. This may fail if we
```

Comment:
Can you try making a program which might trigger a failure perhaps?

Reply:
I've played around with this, but I could give it another go. I might ask them on the IRC and see what they say. I don't think it should block the PR though, would you agree?

Reply:
Can we add an `assert` or `debug_assert` to stop execution if this case does turn out to manifest?

Reply:
An assertion works for me.